src/api/teams/team_repos.rs

```diff
@@ -0,0 +1,114 @@
+use crate::params;
+use crate::{Octocrab, Result};
+use reqwest::StatusCode;
+
+/// Handler for managing a team's repositories through
+/// GitHub's teams API.
+///
+/// Created with [`TeamHandler::repos`]
+///
+/// [`TeamHandler::repos`]: ./struct.TeamHandler.html#method.repos
+pub struct TeamRepoHandler<'octo> {
+    crab: &'octo Octocrab,
+    org: String,
+    team: String,
+}
+
+/// Whether a team manages a repository.
+pub enum ManagesRepo {
+    Yes,
+    No,
+    Err,
+}
+
+impl<'octo> TeamRepoHandler<'octo> {
+    pub(crate) fn new(crab: &'octo Octocrab, org: String, team: String) -> Self {
+        Self { crab, org, team }
+    }
+
+    /// Checks if a team manages a repository.
+    /// ```no_run
+    /// # async fn run() -> octocrab::Result<()> {
+    /// let manages_repo = octocrab::instance()
+    ///     .teams("owner")
+    ///     .repos("team")
+    ///     .manages("owner", "repo")
+    ///     .await?;
+    /// # Ok(())
+    /// # }
+    /// ```
+    pub async fn manages(
+        &self,
+        repo_owner: impl Into<String>,
+        repo_name: impl Into<String>,
+    ) -> Result<ManagesRepo> {
+        let url = format!(
+            "/orgs/{org}/teams/{team}/repos/{owner}/{repo}",
+            org = self.org,
+            team = self.team,
+            owner = repo_owner.into(),
+            repo = repo_name.into(),
+        );
+        let res = self.crab._get(&url, None::<&()>).await?;
+        Ok(match res.status() {
+            StatusCode::NO_CONTENT => ManagesRepo::Yes,
+            StatusCode::NOT_FOUND => ManagesRepo::No,
+            _ => ManagesRepo::Err,
+        })
+    }
+
+    /// Updates a team's permissions for a repository.
+    /// ```no_run
+    /// # async fn run() -> octocrab::Result<()> {
+    /// use octocrab::params;
+    ///
+    /// octocrab::instance()
+    ///     .teams("owner")
+    ///     .repos("team")
+    ///     .add_or_update("owner", "repo", params::teams::Permission::Maintain)
+    ///     .await?;
+    /// # Ok(())
+    /// # }
+    /// ```
+    pub async fn add_or_update(
+        &self,
+        repo_owner: impl Into<String>,
+        repo_name: impl Into<String>,
+        permission: impl Into<Option<params::teams::Permission>>,
+    ) -> Result<()> {
```

Comment:
Does `()` work here? I would assume it would fail to deserialise an empty response, but I'm not entirely sure.

Reply:
Mhh, thinking about it I also think it would fail.

Reply:
The docs in Serde for serde::Deserializer says:
> **unit:** The type of `()` in Rust. It represents an anonymous value containing no data.

As GitHub responds with 204 No Content, I'd expect `Response::text()` in reqwest to return an empty string.

So I set up a Playground test program:
```rust
use serde_json; // 1.0.52

fn main() {
    let v = serde_json::from_str::<()>("");
    println!("{:?}", v);
}
```

Which gave me:
> Err(Error("EOF while parsing a value", line: 1, column: 0))

Maybe a nice future enhancement to `serde_json`? :D
Anyway I'll fix this in the meantime.